src/azure-cli/azure/cli/command_modules/acr/check_health.py

```diff
@@ -72,11 +72,60 @@ def _subprocess_communicate(command_parts, shell=False):
     return output, warning, stderr, succeeded
 
 
+# Retrieve blob identified by digest - GET {url}/v2/{name}/blobs/{digest}
+def _get_blob(login_server, repository_name, digest):
+    return 'https://{}/v2/{}/blobs/{}'.format(login_server, repository_name, digest)
+
+# Checks inputted image or tag for possible matches
+def _check_image_match(cmd,
+                       registry_name,
+                       repository_name,
+                       login_server,
+                       username,
+                       password,
+                       image):
+    from difflib import get_close_matches
+    from .manifest import _get_manifest_path
+    from .repository import(
+        acr_repository_show_tags,
+        _obtain_data_from_registry)
+
+    if 'sha256' not in image:
+        tags = acr_repository_show_tags(
```

Comment:
why list all tags? wouldn't it be better to just check for the given image?

Reply:
@rosanch  In this method, I am not checking for one given image - I am seeing if there is a possible match with the tag that the user gave me (the given image) and comparing it with the list of tags from their repo. To find a possible match, I would need the comprehensive list. Maybe I am not understanding your question?

Reply:
The method _check_image_match is only called if the input is not found (line 489).